forc-pkg/src/pkg.rs

```diff
@@ -1157,6 +1230,29 @@ pub fn fetch_id(path: &Path, timestamp: std::time::Instant) -> u64 {
     hasher.finish()
 }
 
+/// Given an empty or partially completed `graph`, complete the graph.
+///
+/// If the given `manifest` is of type ManifestFile::Workspace resulting graph will have multiple
+/// root nodes, each representing a member of the workspace. Otherwise resulting graph will only
+/// have a single root node, representing the package that is described by the ManifestFile::Package
+fn fetch_graph(
+    member_manifests: &MemberManifestFiles,
+    offline: bool,
+    graph: &mut Graph,
+    manifest_map: &mut ManifestMap,
+) -> Result<HashSet<NodeIx>> {
+    let mut added_nodes = HashSet::default();
+    for member_pkg_manifest in member_manifests.values() {
+        added_nodes.extend(&fetch_pkg_graph(
+            member_pkg_manifest,
+            offline,
+            graph,
+            manifest_map,
+        )?);
+    }
+    Ok(added_nodes)
+}
```

Comment:
Similarly to my comment [here](https://github.com/FuelLabs/sway/pull/3134/files#r1007934240), I think we can likely improve the efficiency quite a bit by re-using `visited` set within `fetch_pkg_graph` to avoid a lot of repeated work, but this seems to work nicely atm so let's leave it for a future PR :+1: